forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -1059,7 +1059,24 @@ else if (defined.startsWith("Triggered") && sa instanceof SpellAbility) {
             }
             final SpellAbility root = ((SpellAbility)sa).getRootAbility();
             Object o = null;
-            if (defParsed.endsWith("Controller")) {
+            if (defParsed.endsWith("OrController")) {
+                String triggeringType = defParsed.substring(9, defParsed.length() - 12);
+                final Object c = root.getTriggeringObject(AbilityKey.fromString(triggeringType));
+                if (c instanceof Player) {
+                    o = c;
+                }
+                if (c instanceof Card) {
```

Comment:
should these be `else if`s?

possibly there's a smooth way to combine all this with `(defParsed.endsWith("Controller"))` to avoid the repeated lines

Reply:
Most likely else ifs, yeah, since they seem to be mutually exclusive